ksql-engine/src/main/java/io/confluent/ksql/util/ExpressionTypeManager.java

```diff
@@ -110,7 +111,7 @@ protected Expression visitCast(
       final ExpressionTypeContext expressionTypeContext
   ) {
     final Type sqlType = node.getType();
-    if (!(sqlType instanceof PrimitiveType)) {
+    if (!(sqlType instanceof PrimitiveType) && !(sqlType instanceof Decimal)) {
```

Comment:
Is it worth having a marker interface? e.g. have `PrimitveType` and `Decimal` implement `Castable` interface?

Also, add decimals to the message in the exception